File: app/src/main/java/com/chaslay/pos/domain/model/StaffPin.kt
package com.chaslay.pos.domain.model

/** Staff POS PIN length (matches backend StaffService + WebPOS). */
const val STAFF_PIN_MIN_LENGTH = 4
const val STAFF_PIN_MAX_LENGTH = 8

fun sanitizeStaffPinInput(raw: String): String =
    raw.filter { it.isDigit() }.take(STAFF_PIN_MAX_LENGTH)

fun isValidStaffPin(pin: String): Boolean {
    val trimmed = pin.trim()
    return trimmed.all { it.isDigit() } &&
        trimmed.length in STAFF_PIN_MIN_LENGTH..STAFF_PIN_MAX_LENGTH
}

File: dashboard/src/lib/staff-pin.ts
/** Staff POS PIN length (matches backend StaffService + Android POS). */
export const STAFF_PIN_MIN_LENGTH = 4;
export const STAFF_PIN_MAX_LENGTH = 8;

export function sanitizeStaffPinInput(raw: string): string {
  return raw.replace(/\D/g, '').slice(0, STAFF_PIN_MAX_LENGTH);
}

export function isValidStaffPin(pin: string): boolean {
  const trimmed = pin.trim();
  return (
    /^\d+$/.test(trimmed) &&
    trimmed.length >= STAFF_PIN_MIN_LENGTH &&
    trimmed.length <= STAFF_PIN_MAX_LENGTH
  );
}
